src/test/java/com/contrastsecurity/maven/plugin/it/stub/FakeContrastAPI.java

```diff
@@ -165,6 +232,13 @@ private static void downloadAgent(final HttpExchange exchange) {
     exchange.close();
```

Comment:
Minor, but we could create another lambda wrapper factory for `HttpHandler` like `cleanupHandler`:

```
private static HttpHandler cleanupHandler(final HttpHandler httpHandler) {
    return exchange -> {
        try {
            discardRequest(exchange);
            httpHandler.handle(exchange);
        } finally {
            exchange.close();
        }
    }
}
```

That might encapsulate the boilerplate a bit.

Reply:
There are three methods that repeat this behavior, but there are six call sites that would need to be decorated. Which do you believe is the lesser of two evils?

Reply:
I like having cleanup encapsulated as its own thing personally (I think it makes it clearer that we need to perform these steps for each request/response), but if you think that’s overkill, I’m fine with leaving as-is as it’s not much duplicated code like you said.

Reply:
OK. I'll make this change and try to reduce any boilerplate as best I can